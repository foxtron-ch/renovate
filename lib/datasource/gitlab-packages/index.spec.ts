import { getPkgReleases } from '..';
import * as httpMock from '../../../test/http-mock';
import { getName } from '../../../test/util';
import { datasource } from './common';

describe(getName(), () => {
  describe('getReleases', () => {
    it('returns package from custom registry', async () => {
      const body = [
        {
          version: '1.0.0',
          created_at: '2020-03-04T12:01:37.000-06:00',
          name: 'mypkg',
        },
        {
          version: 'v1.1.0',
          created_at: '2020-04-04T12:01:37.000-06:00',
          name: 'mypkg',
          _links: {
            web_path: '/user/project1/-/packages/3',
          },
        },
        {
          version: 'v1.1.1',
          created_at: '2020-05-04T12:01:37.000-06:00',
          name: 'mypkg',
        },
        {
          version: 'v2.0.0',
          created_at: '2020-05-04T12:01:37.000-06:00',
          name: 'otherpkg',
        },
      ];
      httpMock
        .scope('https://gitlab.com')
        .get('/api/v4/projects/user%2Fproject1/packages')
        .query({
          package_name: 'mypkg',
          per_page: '100',
        })
        .reply(200, body);
      const res = await getPkgReleases({
        datasource,
        registryUrls: ['https://gitlab.com/user/project1'],
        depName: 'mypkg',
      });
      expect(res).toMatchSnapshot();
      expect(res.releases).toHaveLength(3);
      expect(httpMock.getTrace()).toMatchSnapshot();
    });
  });
});
