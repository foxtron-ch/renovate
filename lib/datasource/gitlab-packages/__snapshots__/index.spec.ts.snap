// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datasource/gitlab-packages/index getReleases returns null for 404 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate, br",
      "host": "gitlab.com",
      "user-agent": "RenovateBot/0.0.0-semantic-release (https://github.com/renovatebot/renovate)",
    },
    "method": "GET",
    "url": "https://gitlab.com/api/v4/projects/user%2Fproject1/packages?package_name=mypkg&per_page=100",
  },
]
`;

exports[`datasource/gitlab-packages/index getReleases returns null for empty 200 OK 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate, br",
      "host": "gitlab.com",
      "user-agent": "RenovateBot/0.0.0-semantic-release (https://github.com/renovatebot/renovate)",
    },
    "method": "GET",
    "url": "https://gitlab.com/api/v4/projects/user%2Fproject1/packages?package_name=mypkg&per_page=100",
  },
]
`;

exports[`datasource/gitlab-packages/index getReleases returns package from custom registry 1`] = `
Object {
  "registryUrl": "https://gitlab.com/user/project1",
  "releases": Array [
    Object {
      "registryUrl": null,
      "releaseTimestamp": "2020-03-04T18:01:37.000Z",
      "version": "1.0.0",
    },
    Object {
      "registryUrl": "https://gitlab.com/user/project1/-/packages/3",
      "releaseTimestamp": "2020-04-04T18:01:37.000Z",
      "version": "v1.1.0",
    },
    Object {
      "registryUrl": null,
      "releaseTimestamp": "2020-05-04T18:01:37.000Z",
      "version": "v1.1.1",
    },
  ],
}
`;

exports[`datasource/gitlab-packages/index getReleases returns package from custom registry 2`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate, br",
      "host": "gitlab.com",
      "user-agent": "RenovateBot/0.0.0-semantic-release (https://github.com/renovatebot/renovate)",
    },
    "method": "GET",
    "url": "https://gitlab.com/api/v4/projects/user%2Fproject1/packages?package_name=mypkg&per_page=100",
  },
]
`;

exports[`datasource/gitlab-packages/index getReleases throws for 5xx 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate, br",
      "host": "gitlab.com",
      "user-agent": "RenovateBot/0.0.0-semantic-release (https://github.com/renovatebot/renovate)",
    },
    "method": "GET",
    "url": "https://gitlab.com/api/v4/projects/user%2Fproject1/packages?package_name=mypkg&per_page=100",
  },
]
`;
